File: apps/agent/schemas/hello.py
"""Hello-world contract mirrors of `packages/shared-schemas/src/hello.ts`.

D3 placeholder — deleted once real listing-pack contracts land.
"""

from pydantic import BaseModel, Field


class HelloRequest(BaseModel):
    message: str = Field(min_length=1, max_length=500)


class HelloResponse(BaseModel):
    message: str
    plan: list[str]
    response: str

File: packages/shared-schemas/src/hello.ts
import { z } from 'zod';

/**
 * D3 placeholder contract — exercised end-to-end to validate web ↔ agent wiring.
 * Will be deleted once real listing-pack contracts replace it.
 */

export const helloRequestSchema = z.object({
  message: z.string().min(1).max(500),
});
export type HelloRequest = z.infer<typeof helloRequestSchema>;

export const helloResponseSchema = z.object({
  message: z.string(),
  plan: z.array(z.string()),
  response: z.string(),
});
export type HelloResponse = z.infer<typeof helloResponseSchema>;
